docs/basics/upgradeability.md

```diff
@@ -122,6 +131,172 @@ Using this pattern, you can introduce other message to your proxy contract.
 Any messages that are not matched in the proxy contract 
 will be forwarded to the specified contract address.
 
+## Delegating execution to foreign Contract Code with `delegate_call`
+
+Similar to proxy-forwarding we can delegate execution to another code hash uploaded on-chain.
+
+### Properties 
+
+- Delegates any call that does not match a selector of itself to another contract.
+- Code is required to uploaded on-chain, but is not required to be instantiated.
+- State is stored in the storage of the original contract which submits the call.
+- Storage layout must be identical between both contract codes.
+
+```
+User ---- tx ---> Contract ----------> Code_v0
+                     ^                     |
+                     |                     |
+                     ⌊_____________________⌋
+                  Storage update is propagated
```

Comment:
I don't think this is the correct intuition, the storage update itself being propogated. It's the code to which the call has been delegated operating on the storage of the calling contract directly

Reply:
I don't know how much detail we want to give on the execution stack without throwing too much theory onto the table.
Nonetheless, I agree that "propagation" is not the most accurate description of the process.

Reply:
We don't need too much detail, but we do need to describe what is happening correctly, and for me "Storage update is propogated" is misleading.